### PR TITLE
fix: Walk through stack to find the correct caller

### PR DESCRIPTION
**Pull request by joeirimpan**

### CODE REVIEW

File: ploggy/base/handler.py [hunk 1]
```diff
@@ -6,6 +11,61 @@
...
+        f = orig_f
+    rv = "(unknown file)", 0, "(unknown function)", None
+    while hasattr(f, "f_code"):
+        co = f.f_code
+        filename = os.path.normcase(co.co_filename)
+        if filename == _srcfile:
+            f = f.f_back
+            continue
+        sinfo = None
+        if stack_info:
```
Comment (joeirimpan) [inline ploggy/base/handler.py:57]: not using stack_info now. but we can expose these as kwargs from the .log func in future like std logging lib

File: ploggy/base/handler.py [hunk 2]
```diff
@@ -6,6 +11,61 @@
...
+    Find the stack frame of the caller so that we can note the source
+    file name, line number and function name.
+
+    ref: https://github.com/python/cpython/blob/main/Lib/logging/__init__.py#L1553
+    """
+    f = currentframe()
+    if f is not None:
+        f = f.f_back
+    orig_f = f
+    while f and stacklevel > 1:
```
Comment (joeirimpan) [inline ploggy/base/handler.py:44]: not using stacklevel as of now. could potentially expose this from .log func in future.

File: ploggy/base/handler.py [hunk 3]
```diff
@@ -18,7 +78,14 @@ class Handler:
...
+            fn, lno, func, _ = find_caller()
+        except ValueError:
+            fn, lno, func = "(unknown file)", 0, "(unknown function)"
+
+        line_ctx = f"{fn}:{lno}:{func}"
+
+        print(self.format(line_ctx, entry), file=self.pipe, flush=True)
 
-    def format(self, entry: Entry) -> str:
+    def format(self, line_ctx: str, entry: Entry) -> str:
```
Comment (joeirimpan) [inline ploggy/base/handler.py:90]: breaking the internal .format syntax

File: ploggy/base/handler.py (hunk 1 above)
Comment (Thunderbottom) [reply ploggy/base/handler.py:57]: yeah makes sense

File: ploggy/base/handler.py (hunk 3 above)
Comment (Thunderbottom) [reply ploggy/base/handler.py:90]: this is alright since it actually gets the right caller

**Review by Thunderbottom — APPROVED** [review]
looks good to me!

**Pull request merged**